tests/cross_functional/kcs/test_monitor_recovery.py

```diff
@@ -529,48 +573,86 @@ def deployments_to_revert(self):
             )
         return to_revert_patches_path, to_revert_mds_path
 
-    def get_ceph_keyrings(self):
+    def get_all_keyring_secrets(self):
         """
-        Gets all ceph and csi related keyring from OCS secrets
+        Get all the keyring secrets
+
+        Returns:
+            list: A list of keyring secrets
 
         """
-        mon_k = get_ceph_caps(["rook-ceph-mons-keyring"])
-        if config.ENV_DATA["platform"].lower() in constants.ON_PREM_PLATFORMS:
-            rgw_k = get_ceph_caps(
-                ["rook-ceph-rgw-ocs-storagecluster-cephobjectstore-a-keyring"]
-            )
-        else:
-            rgw_k = None
-        mgr_k = get_ceph_caps(["rook-ceph-mgr-a-keyring"])
-        mds_k = get_ceph_caps(
-            [
-                "rook-ceph-mds-ocs-storagecluster-cephfilesystem-a-keyring",
-                "rook-ceph-mds-ocs-storagecluster-cephfilesystem-b-keyring",
-            ]
-        )
-        crash_k = get_ceph_caps(["rook-ceph-crash-collector-keyring"])
-        fs_node_k = get_ceph_caps([constants.CEPHFS_NODE_SECRET])
-        rbd_node_k = get_ceph_caps([constants.RBD_NODE_SECRET])
-        fs_provisinor_k = get_ceph_caps([constants.CEPHFS_PROVISIONER_SECRET])
-        rbd_provisinor_k = get_ceph_caps([constants.RBD_PROVISIONER_SECRET])
-
-        keyring_caps = {
-            "mons": mon_k,
-            "rgws": rgw_k,
-            "mgrs": mgr_k,
-            "mdss": mds_k,
-            "crash": crash_k,
-            "fs_node": fs_node_k,
-            "rbd_node": rbd_node_k,
-            "fs_provisinor": fs_provisinor_k,
-            "rbd_provisinor": rbd_provisinor_k,
-        }
-
-        for secret, caps in keyring_caps.items():
-            if caps:
-                with open(f"{self.keyring_dir}/{secret}.keyring", "w") as fd:
-                    fd.write(caps)
-                    self.keyring_files.append(f"{self.keyring_dir}/{secret}.keyring")
+        all_secrets = get_secret_names()
+        keyring_secrets = [keyring for keyring in all_secrets if "keyring" in keyring]
+        return keyring_secrets
+
+    def get_ceph_daemons_keyrings(self):
```

Comment:
this function can be simplified with helpers, osd keyring, other daemon keyrings, and writing keyring to files.